### PR TITLE
Refactor bank's load_and_execute_transaction

### PR DESCRIPTION
**Pull request by dmakarov**
#### Problem

Processing a batch of transactions in bank currently depends on locking the accounts used by the transactions in the batch. This refactoring isolates the locking in a wrapper method, thus preparing for future extraction of the transaction processing code out of bank.

#### Summary of Changes

Add an internal method that does the processing of sanitized transaction batch after the accompanying account locking has been performed in the caller to the new internal method.  Temporarily add a new structure to wrap results of transaction batch processing.

### CODE REVIEW

File: runtime/src/bank.rs [hunk 1]
```diff
@@ -5180,9 +5237,9 @@ impl Bank {
             &self.rc.accounts.accounts_db,
             &self.ancestors,
             sanitized_txs,
-            check_results,
+            check_results.to_owned(),
```
Comment (apfitzge) [inline runtime/src/bank.rs:5240]: This clones the `check_results` which we were not doing before. Why did this become necessary?
I think cloning these should be *relatively* light, but it does involve an Arc clone.

File: runtime/src/bank.rs [hunk 2]
```diff
@@ -5159,10 +5175,51 @@ impl Bank {
...
+            executed_non_vote_transactions_count: sanitized_output
+                .executed_non_vote_transactions_count,
+            executed_with_successful_result_count: sanitized_output
+                .executed_with_successful_result_count,
+            signature_count: sanitized_output.signature_count,
+            error_counters,
+        }
+    }
+
+    #[allow(clippy::type_complexity)]
```
Comment (apfitzge) [inline runtime/src/bank.rs:5204]: don't need this marker anymore since the return type was made into a proper struct

File: runtime/src/bank.rs (hunk 1 above)
Comment (apfitzge) [reply runtime/src/bank.rs:5240]: I think we can just pass check results as an owned vector and avoid this clone

Comment (dmakarov) [reply runtime/src/bank.rs:5240]: a slice was suggested by clippy over a vector. in addition filter_executable_program_accounts called from this method modifies the check_results.

Comment (pgarg66) [inline runtime/src/bank.rs:5239]: We should probably change `load_accounts()` to take `lock_results: &[TransactionCheckResult],` as the argument. That'll remove the need for `to_owned()`. That will require some changes to `NonceFull`. Maybe do that as a separate PR first?

**Comment by dmakarov** [comment]
Updated according to review comments.

File: runtime/src/bank.rs [hunk 3]
```diff
@@ -4416,9 +4431,9 @@ impl Bank {
         self.rc.accounts.accounts_db.set_shrink_paths(paths);
     }
 
-    fn check_age<'a>(
+    fn check_age(
         &self,
-        txs: impl Iterator<Item = &'a (impl core::borrow::Borrow<SanitizedTransaction> + 'a)>,
+        sanitized_txs: &[impl core::borrow::Borrow<SanitizedTransaction>],
```
Comment (pgarg66) [inline runtime/src/bank.rs:4436]: maybe not needed as part of this PR?

File: runtime/src/bank.rs [hunk 4]
```diff
@@ -5158,10 +5174,50 @@ impl Bank {
...
+        sanitized_txs: &[SanitizedTransaction],
+        check_results: &mut [TransactionCheckResult],
+        error_counters: &mut TransactionErrorMetrics,
+        enable_cpi_recording: bool,
+        enable_log_recording: bool,
+        enable_return_data_recording: bool,
+        timings: &mut ExecuteTimings,
+        account_overrides: Option<&AccountOverrides>,
+        log_messages_bytes_limit: Option<usize>,
+        check_time: &mut Measure,
```
Comment (pgarg66) [inline runtime/src/bank.rs:5215]: we can probably not pass it in as parameter. It's only being logged, and accumulated in `timings`. That can be done in the caller.

**Comment by dmakarov** [comment]
Updated removing the check_time parameter and the commit updating check_age signature.

**Review by pgarg66 — APPROVED** [review]
